src/system/sargon_os/sargon_os_profile.rs

```diff
@@ -20,12 +21,12 @@ impl SargonOS {
     /// header
     pub async fn claim_profile(&self, profile: &mut Profile) -> Result<()> {
         debug!("Claiming profile, id: {}", &profile.id());
-        let device_info = self.device_info().await?;
-        Self::claim_provided_profile(profile, device_info.clone());
+        let host_info = self.host_info().await?;
+        Self::claim_provided_profile(profile, host_info.clone().into());
```

Comment:
Host info needs to be translated `into()` device info for claim logic that exists in profile. So we need to translate sargon os information with a language that profile understands (DeviceInfo).